fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/BulkDataUtil.java

```diff
@@ -284,4 +287,37 @@ public static String checkAndValidateJob(Parameters parameters) throws FHIROpera
 
         throw new FHIROperationException("no job identifier is passed");
     }
+
+
+    public static String encryptBatchJobId(String strToEncrypt, SecretKeySpec key) {
+        // Encrypt and UrlEncode the batch job id.
+        if (key == null) {
+            return strToEncrypt;
+        } else {
+            try
+            {
+                Cipher cp = Cipher.getInstance("AES/ECB/PKCS5Padding");
+                cp.init(Cipher.ENCRYPT_MODE, key);
+                return java.net.URLEncoder.encode(Base64.getEncoder().encodeToString(cp.doFinal(strToEncrypt.getBytes("UTF-8"))), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                return strToEncrypt;
```

Comment:
so it will default through to the normal job id?

Reply:
I think we should remove '+' and '=' before URL encoding

Reply:
yes, if the encryption key is not configured in server.xml, then the id will not be encrypted.
en, if remove, how can we bring back the the '+' and '=' before decryption later when receiving the pull request ...

Reply:
changed to use fhir server config.